consensus/hotstuff/voteaggregator/vote_aggregator_v2.go

```diff
@@ -33,15 +37,27 @@ type VoteAggregatorV2 struct {
 var _ hotstuff.VoteAggregatorV2 = &VoteAggregatorV2{}
 
 // NewVoteAggregatorV2 creates an instance of vote aggregator
-func NewVoteAggregatorV2(notifier hotstuff.Consumer, highestPrunedView uint64, committee hotstuff.Committee, voteValidator hotstuff.Validator, signer hotstuff.SignerVerifier) *VoteAggregatorV2 {
-	return &VoteAggregatorV2{
+// NOTE: verifyingProcessorFactory has to be passed outside to abstract out knowledge about Consensus vs Collection
+// cluster running vote aggregation.
+func NewVoteAggregatorV2(log zerolog.Logger, notifier hotstuff.Consumer, highestPrunedView uint64, committee hotstuff.Committee, voteValidator hotstuff.Validator, signer hotstuff.SignerVerifier,
+	verifyingProcessorFactory votecollector.VerifyingVoteProcessorFactory) *VoteAggregatorV2 {
+
+	aggregator := &VoteAggregatorV2{
+		log:               log,
 		notifier:          notifier,
 		highestPrunedView: counters.NewMonotonousCounter(highestPrunedView),
 		committee:         committee,
 		voteValidator:     voteValidator,
 		signer:            signer,
 		unit:              engine.NewUnit(),
 	}
+
+	newCollectorFactoryMethod := func(view uint64) (hotstuff.VoteCollector, error) {
+		return votecollector.NewStateMachine(view, log, aggregator.workerPool, notifier, verifyingProcessorFactory), nil
+	}
+
+	aggregator.collectors = NewVoteCollectors(highestPrunedView, newCollectorFactoryMethod)
```

Comment:
I would expect that it is easier to inject the implementation of `VoteCollectors`. Thereby, we can mock `VoteCollectors` for testing purposes.

Reply:
Let's address this in scope of vote aggregator implementation. I don't want to add/remove too much of fields at this PR.